x/marketmap/keeper/msg_server.go

```diff
@@ -123,6 +173,32 @@ func (ms msgServer) UpdateMarkets(goCtx context.Context, msg *types.MsgUpdateMar
 	return &types.MsgUpdateMarketsResponse{}, ms.k.SetLastUpdated(ctx, uint64(ctx.BlockHeight()))
 }
 
+// verifyMarketAuthoritiesAndExtractContext verifies that the msg-submitter is a market-authority
+// and returns the context for the msg, this method returns an error if the submitter is not a market
+// authority.
+func (ms msgServer) verifyMarketAuthoritiesAndExtractContext(goCtx context.Context, msg interface {
```

Comment:
Why does this function need to return a context?

can't we just pass an sdk.Context into here always? rather than passing a context.Context

Reply:
I wanted to reduce code duplication across the handlers, but I can unwrap context before if u prefer